sympy/solvers/solvers.py

```diff
@@ -228,7 +228,7 @@ def checksol(f, symbol, sol=None, **flags):
                     return False
                 # there are free symbols -- simple expansion might work
                 _, val = val.as_content_primitive()
-                val = expand_mul(expand_multinomial(val))
+                val = simplify(expand_mul(expand_multinomial(val)))
```

Comment:
I think it is too early in the checksol routine to use simplification...perhaps at a later step would be better? Notice that at `attempt=2` we return if `val` is not zero. So if someone wanted minimal simplification I don't think they would want a full simplification attempt at step 1.